File: Node/Euler013.js
#!/usr/bin/env node
const fs = require('fs');

function main() {
  const inFile = "input/Euler013.txt"
  const digits = 10

  const nums = loadFile(inFile)
  const sum = nums.reduce((aggregator, value) => aggregator + value)

  const exponentAdjustment = Math.ceil(Math.log10(sum)) - digits
  const sumAdjustment = Math.pow(10, exponentAdjustment)

  const adjustedSum = Math.round(sum / sumAdjustment)
  console.log(adjustedSum)
}

function loadFile(path) {
  const contents = fs.readFileSync(path, 'utf8');
  values = contents.split("\n").filter((line) => line).map((line) => parseFloat(line) );
  return values
}

if (require.main == module) {
  main();
}

File: Kotlin/Euler013.kt
package Euler013

import java.io.File
import java.lang.Math

fun main() {
  val inFile = "input/Euler013.txt"
  val digits = 10

  val sum = loadFile(inFile).sum()
  val exponentAdjustment = Math.ceil(Math.log10(sum)) - digits
  val sumAdjustment = Math.pow(10.0, exponentAdjustment)
  val firstDigits = (sum / sumAdjustment).toLong()
  println(firstDigits)
}

fun loadFile(path: String): List<Double> {
  var lines: MutableList<Double> = mutableListOf()

  File(path).forEachLine { line -> lines.add(line.toDouble()) }

  return lines
}
